commons/src/main/java/fr/free/nrw/commons/upload/ShareActivity.java

```diff
@@ -208,4 +220,35 @@ public boolean onOptionsItemSelected(MenuItem item) {
         return super.onOptionsItemSelected(item);
     }
 
+    /**
+     * Builds URL with image coords for MediaWiki API calls
+     * Example URL: https://commons.wikimedia.org/w/api.php?action=query&prop=categories|coordinates|pageprops&format=json&clshow=!hidden&coprop=type|name|dim|country|region|globe&codistancefrompoint=38.11386944444445|13.356263888888888&
+     * generator=geosearch&redirects=&ggscoord=38.11386944444445|13.356263888888888&ggsradius=100&ggslimit=10&ggsnamespace=6&ggsprop=type|name|dim|country|region|globe&ggsprimary=all&formatversion=2
+     */
+    public static class UrlBuilder {
+        private static String buildUrl (String coords){
+
+            Uri.Builder builder = Uri.parse("https://commons.wikimedia.org/").buildUpon();
+
+            builder.appendPath("w")
+                    .appendPath("api.php")
+                    .appendQueryParameter("action", "query")
+                    .appendQueryParameter("prop", "categories|coordinates|pageprops")
+                    .appendQueryParameter("format", "json")
+                    .appendQueryParameter("clshow", "!hidden")
+                    .appendQueryParameter("coprop", "type|name|dim|country|region|globe")
+                    .appendQueryParameter("codistancefrompoint", coords)
+                    .appendQueryParameter("generator", "geosearch")
+                    .appendQueryParameter("ggscoord", coords)
+                    .appendQueryParameter("ggsradius", "100")
+                    .appendQueryParameter("ggslimit", "10")
+                    .appendQueryParameter("ggsnamespace", "6")
+                    .appendQueryParameter("ggsprop", "type|name|dim|country|region|globe")
+                    .appendQueryParameter("ggsprimary", "all")
+                    .appendQueryParameter("formatversion", "2");
+
+            return builder.toString();
+        }
+    }
+
 }
```

Comment:
Ah. I tried altering the pom.xml to add the gson dependency with the following code:
'''
      <dependency>
          <groupId>com.google.code.gson</groupId>
          <artifactId>gson</artifactId>
          <version>2.1</version>
          <scope>import</scope>
      </dependency>
'''

But that results in a rather unrelated error popping up:

Error:(7, 24) java: cannot find symbol
  symbol:   class DocumentsContract
  location: package android.provider
Error:(31, 30) java: cannot find symbol
  symbol:   variable DocumentsContract
  location: class fr.free.nrw.commons.upload.FilePathConverter

Worse, when I remove the GSON dependency to bring pom.xml back to its original state (no changes shown on GitHub), that error persists. I guess I should try cloning a fresh copy again...?